src/openfl/utils/_internal/ShaderMacro.hx

```diff
@@ -199,6 +442,249 @@ class ShaderMacro
+				#else
+				var result = "out vec4 fragColor;\n";
+				// var result = "";
+				#end
+				return result;
+			case "310 es":
+				var result = buildGLSLHeaders("300 es");
+				return result;
+			case "320 es":
+				var result = buildGLSLHeaders("310 es");
+				return result;
+
+			case "330":
+				var result = buildGLSLHeaders("320 es");
+				return result;
+
+			case "400":
+				var result = buildGLSLHeaders("330");
+				return result;
+			case "410":
+				var result = buildGLSLHeaders("400");
+				return result;
+			case "420":
+				var result = buildGLSLHeaders("410");
+				return result;
+			case "430":
+				var result = buildGLSLHeaders("420");
+				return result;
+			case "440":
+				var result = buildGLSLHeaders("430");
+				return result;
+			case "450":
+				var result = buildGLSLHeaders("440");
+				return result;
+			case "460":
+				var result = buildGLSLHeaders("450");
+				return result;
+		}
+	}
+
+	private static function buildGLSLExtensions(glExtensions:Array<{name:String, behavior:String}>, glVersion:String,
+			isFragment:Bool):Array<{name:String, behavior:String}>
+	{
+		if (StringTools.endsWith(glVersion, " compatibility")) return glExtensions;
+		if (StringTools.endsWith(glVersion, " core")) return buildGLSLExtensions(glExtensions, StringTools.replace(glVersion, " core", ""), isFragment);
+
+		switch (glVersion)
+		{
+			default:
+				return glExtensions;
+
+			case "100":
+				return glExtensions;
+			case "110":
+				return glExtensions;
+			case "120":
+				return glExtensions;
+			case "130":
+				return glExtensions;
+			case "140":
+				return glExtensions;
+			case "150":
+				return glExtensions;
+
+			case "300 es":
+				var hasSeparateShaderObjects = false;
+				for (extension in glExtensions)
+				{
+					if (extension.name == "GL_ARB_separate_shader_objects") hasSeparateShaderObjects = true;
+					if (extension.name == "GL_EXT_separate_shader_objects") hasSeparateShaderObjects = true;
+				}
+
+				#if desktop
+				if (!hasSeparateShaderObjects)
+				{
+					#if linux
+					return glExtensions.concat([{name: "GL_EXT_separate_shader_objects", behavior: "require"}]);
+					#else
+					return glExtensions.concat([{name: "GL_ARB_separate_shader_objects", behavior: "require"}]);
+					#end
+				}
+				#end
+				return glExtensions;
+
+			case "310 es":
+				var result = buildGLSLExtensions(glExtensions, "300 es", isFragment);
+				return result;
+
+			case "320 es":
+				var result = buildGLSLExtensions(glExtensions, "310 es", isFragment);
+				return result;
+
+			case "330":
+				var result = buildGLSLExtensions(glExtensions, "320 es", isFragment);
+				return result;
+
+			case "400":
+				return glExtensions;
+			case "410":
+				return glExtensions;
+			case "420":
+				return glExtensions;
+			case "430":
+				return glExtensions;
+			case "440":
+				return glExtensions;
+			case "450":
+				return glExtensions;
+			case "460":
+				return glExtensions;
```

Comment:
I don't get why this switch seems so needlessly convoluted. 

- Why have a bunch of cases that behave exactly like the default case?
- why not group similar results using `case "400" | "410"?
- why "310 es" pass "300 es" into `buildGLSLExtensions` and so on for "320 es" and "330"?